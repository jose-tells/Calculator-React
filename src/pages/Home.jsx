import React, { useState } from 'react';
// Redux
import { connect } from 'react-redux';
// Components
import Title from '../components/Title';
import Input from '../components/Input';
// Styles
import '../assets/styles/HomeImports.styl';

const Home = ({ sections }) => {
  const [isOpen, open] = useState(false);

  return (
    <main className='home'>
      <Title />
      <Input
        isHome
        isOpen={isOpen}
        openFn={open}
        functionList={sections}
        placeholderText='Choose an option'
      />
    </main>
  );
};

const mapStateToProps = (state) => {
  return {
    sections: state.sections,
  };
};

export default connect(mapStateToProps, null)(Home);
