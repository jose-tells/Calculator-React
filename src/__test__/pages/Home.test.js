import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Home from '../../pages/Home';
import ProviderMock from '../../__mocks__/ProviderMock';

describe('<Home />', () => {
  test('Home component render', () => {
    // Testing if the component is mounted, just in one layer of depth
    const home = shallow(
      <ProviderMock>
        <Home />
      </ProviderMock>,
    );
    expect(home.length).toEqual(1);
  });
  test('Home component UI snapshot', () => {
    const home = create(
      <ProviderMock>
        <Home />
      </ProviderMock>,
    );
    expect(home.toJSON()).toMatchSnapshot();
  });
});
