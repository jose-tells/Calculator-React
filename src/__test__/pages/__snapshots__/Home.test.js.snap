// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Home component UI snapshot 1`] = `
<main
  className="home"
>
  <header
    className="header"
  >
    <h1
      className="header__title"
    >
      Hi!
      <br />
      What do you want to
      <br />
      calculate?
    </h1>
  </header>
  <form
    action=""
    onSubmit={[Function]}
  >
    <div
      className="home__form isHome"
    >
      <input
        className="home__form--input"
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-left fa-w-10 home__form--icon"
        data-icon="chevron-left"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="home__listBox"
    >
      <ul>
        <a
          className="home__listBox--listItem"
          href="/average"
          onClick={[Function]}
        >
          Average
        </a>
        <a
          className="home__listBox--listItem"
          href="/geo"
          onClick={[Function]}
        >
          Geometric Shapes
        </a>
        <a
          className="home__listBox--listItem"
          href="/median"
          onClick={[Function]}
        >
          Median
        </a>
        <a
          className="home__listBox--listItem"
          href="/mode"
          onClick={[Function]}
        >
          Mode
        </a>
        <a
          className="home__listBox--listItem"
          href="/"
          onClick={[Function]}
        >
          Percentage & Discounts
        </a>
        <a
          className="home__listBox--listItem"
          href="/salary"
          onClick={[Function]}
        >
          Salary Analysis
        </a>
      </ul>
    </div>
  </form>
</main>
`;
